tests/e2e-evm/test/connect.test.ts

```diff
@@ -0,0 +1,50 @@
+import hre from "hardhat";
+import { expect } from "chai";
+import { whaleAddress, userAddress } from "./addresses";
+
+describe("Viem Setup", function () {
+  it("getChainID() matches the configured network", async function () {
+    const publicClient = await hre.viem.getPublicClient();
+
+    const expectedChainId = (() => {
+      switch (hre.network.name) {
+        case "hardhat":
+          return 31337;
+        case "kvtool":
+          return 8888;
+      }
+    })();
+
+    expect(hre.network.config.chainId).to.eq(expectedChainId);
+    expect(await publicClient.getChainId()).to.eq(expectedChainId);
+  });
+
+  it("is configured with whale and user accounts", async function () {
+    const walletClients = await hre.viem.getWalletClients();
+
+    expect(walletClients.length).to.equal(2);
+    expect(walletClients[0].account.address).to.equal(whaleAddress);
+    expect(walletClients[1].account.address).to.equal(userAddress);
+  });
+
+  [
+    { name: "whale", from: whaleAddress, to: userAddress, value: 1n },
```

Comment:
the `name` keys here make me lean more towards having a config object with address, name, etc.

Reply:
Ditto